firmware/hw_layer/debounce.cpp

```diff
@@ -12,35 +12,41 @@
 
 ButtonDebounce* ButtonDebounce::s_firstDebounce = nullptr;
 
-void ButtonDebounce::init (efitimems_t t, brain_pin_e *pin, pin_input_mode_e *mode) {
+/**
+We need to have a separate init function because we do not have the pin or mode in the context in which the class is originally created
+*/
+void ButtonDebounce::init (efitimems_t threshold, brain_pin_e *pin, pin_input_mode_e *mode) {
+   // we need to keep track of whether we have already been initialized due to the way unit tests run.
     if (!initialized) {
-        ButtonDebounce *listItem = s_firstDebounce;
-        if (listItem == nullptr) {
-            s_firstDebounce = this;
-        } else {
-            while (listItem->nextDebounce != nullptr) {
-                listItem = listItem->nextDebounce;
-            }
-            listItem->nextDebounce = this;
-        }
+	// Link us to the list that is used to track ButtonDebounce instances, so that when the configuration changes,
+	//  they can be looped through and updated.
+        nextDebounce = s_firstDebounce;
+        s_firstDebounce = this;
     }
-    threshold = MS2NT(t);
+    m_threshold = MS2NT(threshold);
     timeLast = 0;
-    this->pin = pin;
-    active_pin = *pin;
-    this->mode = mode;
-    active_mode = *mode;
-#ifndef EFI_UNIT_TEST
-    // getInputMode converts from pin_input_mode_e to iomode_t
-    efiSetPadMode("Button", active_pin, getInputMode(active_mode));
-#endif
+    m_pin = pin;
+    m_mode = mode;
+    startConfiguration();
     initialized = true;
 }
 
-void ButtonDebounce::updateConfigurationList () {
+void ButtonDebounce::stopConfigurationList () {
+    ButtonDebounce *listItem = s_firstDebounce;
+    while (listItem != nullptr) {
+        listItem->stopConfiguration();
+        if (listItem->nextDebounce != nullptr) {
```

Comment:
please remove one line here and three lines a bit below

"while (listItem != nullptr)" would have same exact effect, you effectively wrote the same code twice which is not useful

Reply:
done